instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/src/main/java/io/opentelemetry/instrumentation/auto/jaxrs/v2_0/AbstractRequestContextInstrumentation.java

```diff
@@ -74,39 +73,35 @@ public AbstractRequestContextInstrumentation() {
   }
 
   public static class RequestFilterHelper {
-    public static SpanWithScope createOrUpdateAbortSpan(
+    public static Span createOrUpdateAbortSpan(
         ContainerRequestContext context, Class<?> resourceClass, Method method) {
 
       if (method != null && resourceClass != null) {
         context.setProperty(JaxRsAnnotationsTracer.ABORT_HANDLED, true);
-        Span parent = BaseTracer.getCurrentServerSpan();
-        Span span = TRACER.getCurrentSpan();
+        Span serverSpan = BaseTracer.getCurrentServerSpan();
+        Span currentSpan = TRACER.getCurrentSpan();
 
-        if (span == null) {
-          span = TRACER.startSpan(resourceClass, method);
-          return new SpanWithScope(span, currentContextWith(span));
+        if (currentSpan == null || currentSpan == serverSpan) {
```

Comment:
why checking `currentSpan == serverSpan`?

Reply:
I should have added a comment here -- will fix it.

I've found out that `DefaultRequestContextInstrumentation` and implementation-specific `RequestContextInstrumentation`s (e.g. `Resteasy31RequestContextInstrumentation`) worked differently when executed in different order.
* In case of default -> impl order: the default instrumentation has always created a span, but without adding the `ABORT_HANDLED` marker, so the impl instrumentation also ran; `span != null` so only the span names were updated.
* In case of impl -> default: `span` was also not null (because the servlet span would be the current span), so no new span was started here; `ABORT_HANDLED` marker was added though. The default instrumentation did not run; in this case we were missing one span.